### PR TITLE
Docs listening: Overhauled downgrade docs

### PR DESCRIPTION
**Pull request by cwarnermm**

### CODE REVIEW

**Review by isacikgoz — CHANGES_REQUESTED** [review]
Thanks @cwarnermm for raising the changes. This PR requires some changes though.

File: source/upgrade/downgrading-mattermost-server.rst [hunk 1]
```diff
@@ -4,49 +4,46 @@ Downgrade Mattermost Server
...
+
+2. Carefully review the Mattermost changelog for the version you are downgrading to in order to understand any potential issues or incompatibilities.
+
+3. Verify the current schema version of your database using the :ref:`mattermost db version <manage/command-line-tools:mattermost db version>` command.
+
+Perform the downgrade
+---------------------
+
+1. Stop the Mattermost service to ensure that no data is being written to the database during the downgrade process.
+2. Downgrade the application by replacing the current Mattermost application binary with the version you want to downgrade to. Make sure to use the binary of the target version.
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/upgrade/downgrading-mattermost-server.rst]: Really appreciate your help with this, @isacikgoz!

File: source/upgrade/downgrading-mattermost-server.rst [hunk 2]
```diff
@@ -4,49 +4,47 @@ Downgrade Mattermost Server
...
+
+1. Back up your data: Ensure you have a full backup of your database and Mattermost application files. This is crucial in case you need to revert any changes.
+
+  a. Back up your database using your organization’s standard procedures for backing up the database.
+
+  b. Back up your application by copying into an archive folder (e.g. ``mattermost-back-YYYY-MM-DD-HH-mm``). Ensure to copy your Mattermost configuration files and any other necessary application files.
+
+2. Carefully review the Mattermost changelog for the version you are downgrading to in order to understand any potential issues or incompatibilities.
+
+3. Verify the current schema version of your database using the :ref:`mattermost db version <manage/command-line-tools:mattermost db version>` command.
```
Comment (isacikgoz) [inline source/upgrade/downgrading-mattermost-server.rst:28]: Reading it again, we can elaborate this a bit further:

```suggestion
3. Verify the current schema version of your database using the :ref:`mattermost db version --all<manage/command-line-tools:mattermost db version>` command. Also, if you are not sure about the target schema, you can verify the target schema version (ie. applied migrations) by checking the public `GitHub repository  <https://github.com/mattermost/mattermost/blob/master/server/channels/db/migrations/migrations.list>`_ (please select the tag for desired version).
```

**Review by isacikgoz — APPROVED** [review]
Thanks @cwarnermm 💪 I just suggested a change to elaborate even more. Good go after that :)

**Pull request merged**